sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_schema_registry_client.py

```diff
@@ -136,13 +153,18 @@ def get_schema(self, schema_id, **kwargs):
                 :caption: Get schema by id.
 
         """
-        return self._generated_client.schema.get_by_id(
-            schema_id,
-            cls=_parse_response_schema,
-            **kwargs
-        )
-
-    def get_schema_id(self, schema_group, schema_name, serialization_type, schema_content, **kwargs):
+        try:
+            return self._id_to_schema[schema_id]
+        except KeyError:
+            schema = self._generated_client.schema.get_by_id(
+                schema_id, cls=_parse_response_schema, **kwargs
+            )
+            self._id_to_schema[schema_id] = schema
```

Comment:
I'm wondering if the `self._description_to_properties` cache could also be updated in the `get_schema`?
seems every piece of info needed for building an entry in the `self._description_to_properties` could be extracted from the schema.

and if doing such we could get back the `_add_cache` method?

Reply:
I think in this case, the other `_description_to_properties` cache cannot be updated at the same time b/c the other info required to get description (schema_group, schema_name, serialization_type) are all not available here. The only pieces of info available are schema_content and schema_properties. I'm not sure whether it's possible to update that dict here, unless you have a different idea?

Reply:
now that you mention it though, I think it does seem like the other cache needs to be updated somehow, to remain consistent with this one. How do you feel about a `_id_to_description` cache? (which can be used to update the `description_to_properties` cache)

Reply:
correct me if I'm wrong, the SchemaProperties is an ivar of the Schema object, so I think the schema_group and schema_name could be extracted from `schema.schema_properties`:

`schema.schema_properties.schema_group`,  `schema.schema_properties.schema_name`

Reply:
so those two are the only ivars that are not in SchemaProperties, unfortunately. Here is the list of ivars for SchemaProperties:
    :ivar schema_id: References specific schema in registry namespace.
    :type schema_id: str
    :ivar location: URL location of schema, identified by schema group, schema name, and version.
    :type location: str
    :ivar location_by_id: URL location of schema, identified by schema ID.
    :type location_by_id: str
    :ivar serialization_type: Serialization type for the schema being stored.
    :type serialization_type: str
    :ivar version: Version of the returned schema.
    :type version: int

Reply:
my bad for not checking the class

I'm good with an extra `_id_to_description` cache for now, but we definitely should revisit this later.
I don't feel good with the three asymmetry cache maps.

Reply:
agreed, will write add it to the design page

Reply:
final 'offline' conversation:
- decided to not add another asymmetric map, since it would definitely decrease comprehensibility
- for now, to avoid this, we can follow the JS approach, which was to:
1) update both maps for the [`getSchemaId` method](https://github.com/Azure/azure-sdk-for-js/blob/934968168f7464d5a5aa8e8717d1acb7cc77dcd0/sdk/schemaregistry/schema-registry/src/schemaRegistryClient.ts#L125
2) update only the `idToSchemaMap` in the [`getSchemaById` method](https://github.com/Azure/azure-sdk-for-js/blob/934968168f7464d5a5aa8e8717d1acb7cc77dcd0/sdk/schemaregistry/schema-registry/src/schemaRegistryClient.ts#L155)